src/main/pileup.js

```diff
@@ -80,6 +80,37 @@ function create(elOrId: string|Element, params: PileupParams): Pileup {
       ReactDOM.render(<Root referenceSource={referenceTrack.source}
                             tracks={vizTracks}
                             initialRange={params.range} />, el);
+
+  //if the element doesn't belong to document DOM observe DOM to detect
+  //when it's attached
+  var observer = null;
+  if (!document.contains(el)) {
+    observer = new MutationObserver(function(mutations) {
+      mutations.forEach(function(mutation) {
+        if (mutation.type === 'childList') {
+          var added= false;
+          for (var i=0; i<mutation.addedNodes.length;i++) {
+            if (mutation.addedNodes[i]===el) {
+              added = true;
+            }
+          }
+          if (added) {
+            if (reactElement) {
+              reactElement.setState({updateSize:true});
+            } else {
+              throw 'ReactElement was not initialized properly';
+            }
+          }
+        }
+      });
+    });
+    // configuration of the observer:
+    var config = { attributes: true, childList: true, characterData: true, subtree: true };
+
+    // start observing document
+    observer.observe(document, config);
```

Comment:
Fair enough. If you don't have any performance issues in your complex application after this change, it is unlikely that we will have that for other simpler applications. Thanks for the pointer!